pwnlib/util/misc.py

```diff
@@ -248,17 +254,23 @@ def run_in_new_terminal(command, terminal = None, args = None):
 
     log.debug("Launching a new terminal: %r" % argv)
 
-    pid = os.fork()
-
-    if pid == 0:
-        # Closing the file descriptors makes everything fail under tmux on OSX.
-        if platform.system() != 'Darwin':
-            devnull = open(os.devnull, 'rwb')
-            os.dup2(devnull.fileno(), 0)
-            os.dup2(devnull.fileno(), 1)
-            os.dup2(devnull.fileno(), 2)
-        os.execv(argv[0], argv)
-        os._exit(1)
+    stdin = stdout = stderr = open(os.devnull, 'rwb')
+    if terminal == 'tmux':
+        stdout = subprocess.PIPE
+
+    p = subprocess.Popen(argv, stdin=stdin, stdout=stdout, stderr=stderr)
+
+    if terminal == 'tmux':
+        out, _ = p.communicate()
+        pid = int(out)
+    else:
+        pid = p.pid
+
+    if kill_at_exit:
+        if terminal == 'tmux':
+            atexit.register(lambda: os.kill(pid, signal.SIGTERM))
```

Comment:
This could kill the wrong process on PID reuse, but unfortunately these are the APIs we have to work with. Hopefully PID reuse is relatively unlikely to happen during the relatively short lifetime a pwntools script.